lib/PuppeteerSharp/Launcher.cs

```diff
@@ -198,6 +224,24 @@ private void CleanUp()
             _chromeProcess?.RemoveExitedEvent();
         }
 
+        private async Task AfterProcessExit()
+        {
+            if (!IsChromeClosed)
+            {
+                IsChromeClosed = true;
+
+                if (_waitForChromeToClose.Task.Status != TaskStatus.RanToCompletion)
+                {
+                    _waitForChromeToClose.SetResult(true);
+                }
+
+                if (_temporaryUserDataDir != null)
+                {
+                    await Task.Factory.StartNew(path => Directory.Delete((string)path, true), _temporaryUserDataDir);
```

Comment:
line 270 calls this method
on line 275 we delete the directory, so either this line or 275 will fail since one of them will delete the directory

Reply:
Good catch. The problem I was having is that the process `Exited` event was not being called, your `WaitForExit` did the trick.

This PR is ready to review again. I'll implement IDisposables in another PR

Reply:
I added some notes.
you should notice that the close operation is `async` as oppose to `IDisposables.Dispose` which it `void`

Reply:
@Meir017 yeah. We'd need to `CloseAsync().GetAwaiter().GetResult()`, do you have a better idea?

Reply:
should good enough for now.
what we need is - https://github.com/dotnet/roslyn/issues/114